### PR TITLE
Sequence Tagger Decoder re-implemented in Torch.

### PR DESCRIPTION
**Pull request by justinxzhao**
The sequence tagger decoder functionality is complete compared to v0.4 functionality.

All sequence-related unit tests and integration tests are now re-enabled, except for those that use sampled softmax loss (still TODO).

I've also added a unit test for the sequence tagger decoder itself, which could be a useful skeleton for other decoder specific unit tests.

### CODE REVIEW

File: tests/integration_tests/test_graph_execution.py [hunk 1]
```diff
@@ -69,33 +68,17 @@ def test_experiment_multiple_seq_seq(csv_filename, output_features):
...
-@pytest.mark.skip(reason="Issue #1333: Sequence output features.")
 @pytest.mark.distributed
-@pytest.mark.parametrize("dec_beam_width", [3])
-@pytest.mark.parametrize("dec_attention", ["bahdanau"])
-@pytest.mark.parametrize("dec_cell_type", ["lstm"])
-@pytest.mark.parametrize("enc_cell_type", ["lstm"])
+@pytest.mark.parametrize("dec_cell_type", ["rnn"])
+@pytest.mark.parametrize("enc_cell_type", ["rnn"])
 @pytest.mark.parametrize("enc_encoder", ["embed"])
-def test_sequence_generator(enc_encoder, enc_cell_type, dec_cell_type, dec_attention, dec_beam_width, csv_filename):
```
Comment (w4nderlust) [inline tests/integration_tests/test_graph_execution.py:79]: maybe add a todo here to remember to add attention and beam search back?

File: tests/integration_tests/test_sequence_features.py [hunk 2]
```diff
@@ -79,34 +77,29 @@ def setup_model_scaffolding(raw_df, input_features, output_features):
...
+        ((128, 10, TEST_STATE_SIZE), None),
+        ((128, 10, TEST_STATE_SIZE), ((128, TEST_STATE_SIZE), (128, TEST_STATE_SIZE))),
+        ((128, 10, TEST_STATE_SIZE), ((128, TEST_STATE_SIZE),)),
+    ],
 )
 def test_sequence_decoders(
     dec_cell_type,
-    dec_attention,
-    dec_beam_width,
-    dec_num_layers,
```
Comment (w4nderlust) [inline tests/integration_tests/test_sequence_features.py:109]: add todo to reintroduce these?

Comment (justinxzhao) [reply tests/integration_tests/test_sequence_features.py:109]: Done. This test really should belong in test_sequence_decoders.py unit tests, and tests should certainly be added when reintroduce beam search, multi-layers, and attention.

File: tests/integration_tests/test_graph_execution.py (hunk 1 above)
Comment (justinxzhao) [reply tests/integration_tests/test_graph_execution.py:79]: I've gone ahead and removed this particular test completely as it's highly overlapping the test we already have in run_experiment.

I'm planning to test all of these aspects of decoder within test_sequence_generator unit tests.

**Comment by justinxzhao** [comment]
Since this is mostly a for-parity change, to re-enable tests, with all checks passing, I'll go ahead and merge it to master to unblock subsequent PRs.

If there are any additional comments, I'm more than happy to address them in follow-ups!

**Pull request merged**